service/src/main/java/bio/terra/externalcreds/controllers/OpenApiConverters.java

```diff
@@ -75,11 +76,15 @@ public static LinkInfo convert(LinkedAccount linkedAccount) {
           .authenticated(linkedAccount.isAuthenticated());
     }
 
-    public static SshKeyPair convert(SshKeyPairInternal sshKeyPairInternal) {
+    public static SshKeyPair convert(
+        SshKeyPairInternal sshKeyPairInternal, KmsEncryptDecryptHelper encryptDecryptHelper) {
       return new SshKeyPair()
           .externalUserEmail(sshKeyPairInternal.getExternalUserEmail())
           .publicKey(sshKeyPairInternal.getPublicKey())
-          .privateKey(new String(sshKeyPairInternal.getPrivateKey(), StandardCharsets.UTF_8));
+          .privateKey(
+              new String(
+                  encryptDecryptHelper.decryptSymmetric(sshKeyPairInternal.getPrivateKey()),
+                  StandardCharsets.UTF_8));
```

Comment:
all the encryption happens in the service, why is the decryption here? Or vice versa? Why not keep is all in the sam place?

Reply:
yeah i tried both approaches and leave it with the minimal change to see what you would like..

if i do the decryption in the service, the sshKeyPairinternal will sometimes hold encrypted key (going in to the database) and sometimes hold decrypted key (going out).. or we can just convert to sshkeypair in the service?